[BUG] Immediate crash when not using network-manager

**Describe the bug**

The app crashes immediately after configuration when not using a network-manager.

**To Reproduce**

Launch go-hass-agent binary (version 0.0.2)

**Expected behavior**

It should warn network-manager is not reachable/available.


**Logs**
A log will be very helpful to look into this bug report. To get the log:

```
4:54PM INF Starting agent.
4:54PM WRN No suitable existing config found! Starting new registration process
2023/04/12 16:54:42 systray error: failed to register our icon with the notifier watcher (maybe no tray is running?): The name org.kde.StatusNotifierWatcher was not provided by any .service files
4:54PM INF Looking for Home Assistant instances on the network...
4:55PM WRN Unsupported desktop/window environment.
4:55PM ERR Unable to execute org.freedesktop.NetworkManager.GetDevices on org.freedesktop.NetworkManager (args: ) error="The name org.freedesktop.NetworkManager was not provided by any .service files"
panic: interface conversion: interface {} is nil, not []dbus.ObjectPath

goroutine 164 [running]:
github.com/joshuar/go-hass-agent/internal/device.NetworkUpdater({0xfd3780, 0xc001ceecf0}, 0xc001b4b860)
	/home/grego/go/pkg/mod/github.com/joshuar/go-hass-agent@v0.0.2/internal/device/network_linux.go:349 +0x625
github.com/joshuar/go-hass-agent/internal/agent.(*Agent).tracker.func3(0xc000158dc0?)
	/home/grego/go/pkg/mod/github.com/joshuar/go-hass-agent@v0.0.2/internal/agent/agent.go:162 +0x6a
created by github.com/joshuar/go-hass-agent/internal/agent.(*Agent).tracker
	/home/grego/go/pkg/mod/github.com/joshuar/go-hass-agent@v0.0.2/internal/agent/agent.go:160 +0x2ce
```



**Desktop (please complete the following information):**
 - OS: Archlinux
 - Version 0.0.2

--- a/internal/device/network_linux.go
+++ b/internal/device/network_linux.go
@@ -340,13 +340,22 @@ func NetworkUpdater(ctx context.Context, status chan interface{}) {
 	deviceAPI, deviceAPIExists := FromContext(ctx)
 	if !deviceAPIExists {
 		log.Debug().Caller().
-			Msg("Could not connect to DBus to monitor batteries.")
+			Msg("Could not connect to DBus to monitor network.")
 		return
 	}
 
-	deviceList := deviceAPI.GetDBusData(
+	myDeviceList := deviceAPI.GetDBusData(
 		systemBus, dBusDest, dBusPath,
-		"org.freedesktop.NetworkManager.GetDevices", "").([]dbus.ObjectPath)
+		"org.freedesktop.NetworkManager.GetDevices", "")
+
+	if myDeviceList == nil {
+		log.Debug().Caller().
+			Msg("Could not list devices from network manager")
+		return
+	}
+
+	deviceList := myDeviceList.([]dbus.ObjectPath)
+
 	if len(deviceList) > 0 {
 		for _, device := range deviceList {
 			conn := deviceActiveConnection(ctx, device)

